providers/amazon/src/airflow/providers/amazon/aws/hooks/mwaa.py

```diff
@@ -56,30 +64,78 @@ def invoke_rest_api(
 
         :param env_name: name of the MWAA environment
         :param path: Apache Airflow REST API endpoint path to be called
-        :param method: HTTP method used for making Airflow REST API calls
+        :param method: HTTP method used for making Airflow REST API calls: 'GET'|'PUT'|'POST'|'PATCH'|'DELETE'
         :param body: Request body for the Apache Airflow REST API call
         :param query_params: Query parameters to be included in the Apache Airflow REST API call
+        :param only_use_web_login: If True, only the web login method is used without trying boto's
+            invoke_rest_api first
         """
-        body = body or {}
+        # Filter out keys with None values because Airflow REST API doesn't accept requests otherwise
+        body = {k: v for k, v in body.items() if v is not None} if body else {}
+        query_params = query_params or {}
         api_kwargs = {
             "Name": env_name,
             "Path": path,
             "Method": method,
-            # Filter out keys with None values because Airflow REST API doesn't accept requests otherwise
-            "Body": {k: v for k, v in body.items() if v is not None},
-            "QueryParameters": query_params if query_params else {},
+            "Body": body,
+            "QueryParameters": query_params,
         }
+
+        if only_use_web_login:
+            return self._invoke_rest_api_using_web_login(**api_kwargs)
+
         try:
-            result = self.conn.invoke_rest_api(**api_kwargs)
+            response = self.conn.invoke_rest_api(**api_kwargs)
             # ResponseMetadata is removed because it contains data that is either very unlikely to be useful
             # in XComs and logs, or redundant given the data already included in the response
-            result.pop("ResponseMetadata", None)
-            return result
+            response.pop("ResponseMetadata", None)
+            return response
+
         except ClientError as e:
-            to_log = e.response
-            # ResponseMetadata and Error are removed because they contain data that is either very unlikely to
-            # be useful in XComs and logs, or redundant given the data already included in the response
-            to_log.pop("ResponseMetadata", None)
-            to_log.pop("Error", None)
-            self.log.error(to_log)
-            raise e
+            if e.response["Error"]["Code"] == "AccessDeniedException":
+                self.log.info(
+                    "Access Denied, possibly due to missing airflow:InvokeRestApi in IAM policy. "
+                    "Trying again using web login..."
+                )
+                return self._invoke_rest_api_using_web_login(**api_kwargs)
```

Comment:
What if this throws?

Reply:
I assumed the logged `HTTPError` from `raise_for_status()` include `detail` but I just tested and it is not included so I'll add additional logging for this as well. Also, since there's no `ResponseMetadata` in the response, it doesn't need to be removed before logging it, unlike `ClientError`

Reply:
Fair enough, I wasn't necessarily worried about the logging, more that you'll just have an unhandled exception at that point. But if that's okay, and you're happy with that bubbling up on it's own with the messaging it has, then all good :+1:

Reply:
Yea, I think it would provide enough context in logs (it would look very similar to the ClientError in the other case) and make the operator fail when it should be failing anyway